packages/region/src/services/region-module.ts

```diff
@@ -117,78 +115,31 @@ export default class RegionModuleService<
     data: CreateRegionDTO[],
     @MedusaContext() sharedContext: Context = {}
   ): Promise<Region[]> {
-    let currencies = await this.currencyService_.list(
-      { code: data.map((d) => d.currency_code.toLowerCase()) },
-      {},
-      sharedContext
-    )
-
-    const countriesInDb = await this.countryService_.list(
-      { iso_2: data.map((d) => d.countries).flat() },
-      { select: ["iso_2", "region_id"] },
-      sharedContext
-    )
+    let normalizedInput = this.normalizeInput(data)
 
-    const countriesInDbMap = new Map<string, Country>(
-      countriesInDb.map((c) => [c.iso_2, c])
-    )
-
-    const currencyMap = new Map(
-      currencies.map((c) => [c.code.toLowerCase(), c])
+    const validations = [
+      this.validateCurrencies(
+        normalizedInput.map((r) => r.currency_code),
+        sharedContext
+      ),
+      this.validateCountries(
+        normalizedInput.map((r) => r.countries ?? []).flat(),
+        sharedContext
+      ),
+    ] as const
+
+    // Assign the full country object so the ORM updates the relationship
+    const [, dbCountries] = await Promise.all(validations)
+    const dbCountriesMap = asMap(dbCountries, "iso_2")
```

Comment:
suggestion: we could create an inline map here wdyt? not sure the util is very useful
```ts
const dbCountriesMap = new Map(dbCountries.map(d => [d.id, d]))
```

Reply:
Yeah I agree, I actually didn't know new Map takes a tuple 👍